distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso

```diff
@@ -530,15 +540,20 @@ type Array
        in the input array.
 
        Arguments:
-       - predicate: A function that takes an element and returns a boolean
-         value.
+       - condition: A `Filter_Condition` or a function that takes an element and
+         returns a boolean value.
```

Comment:
Maybe simplify "a function that takes an element and returns a boolean value" to "predicate function" everywhere.

Reply:
I was wondering if `predicate` is a well known term for non-programmers?

Since I'm not a native speaker - I have no idea. What do you think? 

From my experience of the Polish counterpart `predykat` - it is something that shows up rather only in computer science, mathematical logic or maybe philosophy literature. I would not assume someone with e.g. purely financial background must be familiar with that term.

Reply:
I'm happy to rephrase it, but I don't like "test each element against returning a `Boolean`."

I like `test each element` - that part sounds OK.

But `test ... against returning` - to me it sounds like we are testing collection elements to see if each _element is itself a function that may return a boolean_.

(Test something against doing X sounds to me like test if that thing does X - in our case returns boolean).

So IMO that phrasing is also confusing.

Reply:
IMO we should either have `function returning a boolean` or a `predicate` - having both is a pleonasm. (Although I appreciate the SEO :))

Reply:
Settling for:
```
       - condition: A `Filter_Condition` or a predicate function to test each
         element.
```